### PR TITLE
Zap logging output format parameter support

### LINKED ISSUE #977: Add option to output debug controller logs as JSON
**Describe the solution you'd like**
Add option to write logging-operator debug logs as JSON when using `-verbose=true`

**Additional informations**
The verbose flag https://github.com/banzaicloud/logging-operator/blob/4218d98bb104b816cf634519103fd53848a49f91/main.go#L91 enables zap's Development configurations, which changes the format to console https://github.com/uber-go/zap/blob/6f34060764b5ea1367eecda380ba8a9a0de3f0e6/config.go#L165

**Proposed solution**
Add a loglevel flag that allows keeping the production configuration with a specific loglevel instead of enabling zap's  Developpement configuration .

### PR DESCRIPTION
**Pull request by ahma**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #977
| License         | Apache 2.0

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -80,6 +81,7 @@ func main() {
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
 	flag.BoolVar(&verboseLogging, "verbose", false, "Enable verbose logging")
+	flag.StringVar(&loggingOutputFormat, "output-format", "", "Logging output format json, console")
```
Comment (aslafy-z) [inline main.go:84]:
```suggestion
	flag.StringVar(&loggingOutputFormat, "output-format", "", "Logging output format (json, console)")
```

**Review by aslafy-z — COMMENTED** [review]
See my comments

**Review by bshifter — APPROVED** [review]
lgtm

**Review by aslafy-z — APPROVED** [review]

File: main.go [hunk 2]
```diff
@@ -90,7 +92,22 @@ func main() {
...
+		case "json":
+			encoder := zap.JSONEncoder()
+			encoder(o)
+		case "console":
+			encoder := zap.ConsoleEncoder()
+			encoder(o)
+		case "":
+			break
+		default:
+			fmt.Printf("invalid encoder value \"%s\"", loggingOutputFormat)
```
Comment (siliconbrain) [inline main.go:106]:
```suggestion
			fmt.Printf("invalid encoder value %q\n", loggingOutputFormat)
```

**Review by siliconbrain — APPROVED** [review]
lgtm

**Pull request merged**